plumbing/format/packfile/decoder.go

```diff
@@ -101,8 +106,9 @@ func NewDecoderForType(s *Scanner, o storer.EncodedObjectStorer,
 	}
 
 	return &Decoder{
-		s: s,
-		o: o,
+		s:              s,
+		o:              o,
+		DeltaBaseCache: cacheObject,
```

Comment:
can you make DeltaBaseCache private?

Reply:
I've addressed these issues in the following commits.